core/base/slot_chain.go

```diff
@@ -157,14 +158,11 @@ func (sc *SlotChain) RangeRuleCheckSlot(f func(slot RuleCheckSlot)) {
 	}
 }
 
-func (sc *SlotChain) AddRuleCheckSlotFirst(s RuleCheckSlot) {
-	ns := make([]RuleCheckSlot, 0, len(sc.ruleChecks)+1)
-	ns = append(ns, s)
-	sc.ruleChecks = append(ns, sc.ruleChecks...)
-}
-
-func (sc *SlotChain) AddRuleCheckSlotLast(s RuleCheckSlot) {
+func (sc *SlotChain) AddRuleCheckSlot(s RuleCheckSlot) {
 	sc.ruleChecks = append(sc.ruleChecks, s)
+	sort.Slice(sc.ruleChecks, func(i, j int) bool {
```

Comment:
Maybe we need unit test to cover rank result.

Reply:
There are already related unit tests like `TestSlotChain_AddStatPrepareSlot`. Is is sufficient?

Reply:
Okay for me.